src/zenml/cli/pipeline.py

```diff
@@ -125,37 +130,60 @@ def list_pipelines(**kwargs: Any) -> None:
 
 @pipeline.command("delete")
 @click.argument("pipeline_name_or_id", type=str, required=True)
+@click.option(
+    "--version",
+    "-v",
+    help="Optional pipeline version.",
+    type=str,
+    required=False,
+)
 @click.option(
     "--yes",
     "-y",
     is_flag=True,
     help="Don't ask for confirmation.",
 )
-def delete_pipeline(pipeline_name_or_id: str, yes: bool = False) -> None:
+def delete_pipeline(
+    pipeline_name_or_id: str, version: Optional[str] = None, yes: bool = False
+) -> None:
     """Delete a pipeline.
 
     Args:
         pipeline_name_or_id: The name or ID of the pipeline to delete.
+        version: The version of the pipeline to delete.
         yes: If set, don't ask for confirmation.
     """
     cli_utils.print_active_config()
 
+    version_suffix = (
+        f" (version {version})" if version else " (latest version)"
+    )
+
     if not yes:
         confirmation = cli_utils.confirmation(
             f"Are you sure you want to delete pipeline "
-            f"`{pipeline_name_or_id}`? This will change all existing runs of "
-            "this pipeline to become unlisted."
+            f"`{pipeline_name_or_id}{version_suffix}`? This will change all "
```

Comment:
`f0ab0295-e712-4cc0-a6c0-74931c11246d (latest version)` seems a bit weird to me, perhaps we can move the UUID checks from line 173 up and then only print the version info if it's not an ID?

Reply:
Makes sense, will update it